core/src/com/unciv/ui/audio/MusicController.kt

```diff
@@ -476,4 +483,52 @@ class MusicController {
         getFile(musicFallbackLocalName).exists()
 
     //endregion
+
+    //region city sounds
+
+    private fun getSoundFolders() = sequence {
+        val visualMods = UncivGame.Current.settings.visualMods
+        val mods = UncivGame.Current.gameInfo!!.gameParameters.getModsAndBaseRuleset()
+        yieldAll(
+            (visualMods + mods).asSequence()
+                .map { getFile("mods")
+                    .child(it).child("sounds") }
+        )
+        yield(getFile("sounds"))
+    }
+
+    private fun getSoundFile(fileName: String): FileHandle? = getSoundFolders()
+        .filter { it.exists() && it.isDirectory }
+        .flatMap { it.list().asSequence() }
+        // ensure only normal files with common sound extension
+        .filter { it.exists() && !it.isDirectory && it.extension() in fileExtensions }
+        .firstOrNull { it.name().contains(fileName) }
+
+    fun playCitySound(city: CityInfo) {
+        if (playingCitySound != null)
+            stopCitySound()
+        try {
+            val file = FileHandle(getSoundFile(city.civInfo.getEra().citySound).toString())
+            playingCitySound = Gdx.audio.newMusic(file)
+
+            playingCitySound?.volume = UncivGame.Current.settings.citySoundsVolume
+            playingCitySound?.isLooping = true
+            playingCitySound?.play()
+        } catch (ex: Throwable) {
+            playingCitySound?.dispose()
+            Log.error("Error while playing city sound: ", ex)
+        }
+    }
+
+    fun stopCitySound() {
+        try {
+            if (playingCitySound?.isPlaying == true)
+                playingCitySound?.stop()
+        } catch (ex: Throwable) {
+            playingCitySound?.dispose()
+            Log.error("Error while stopping city sound: ", ex)
+        }
+    }
+
+    //endregion
```

Comment:
ahhhhh 
I reverted that commit back. But if I remove `: MusicController()` from `class CitySoundPlayer` I can't install the hooks without those 2 methods which I don't really know what to put in them.. leave them empty? And actually I shouldn't install the hooks as that would stop the music...

But if I don't install the hooks wouldn't it create the issues mentioned by SomeTroglodyte ?

Reply:
> why would it?

to avoid these errors

https://github.com/yairm210/Unciv/blob/9d5b3d4e4a9979d7b52bf0d9580309c672f960ea/desktop/src/com/unciv/app/desktop/HardenGdxAudio.kt#L30

Or is the new audio already installed and that would catch errors which happen in here too?